lib/src/server.dart

```diff
@@ -170,7 +173,8 @@ class Server {
         'result': result,
         'id': request['id']
       };
-    }).catchError((error, stackTrace) {
+    } catch (e, stackTrace) {
+      var error = e;
```

Comment:
In that case, I'd just return the serialized exception directly in the `is! RpcException` case rather than re-assigning here. You could also pull out `if (!request.containsKey('id')) return null` to the beginning of the `catch` block.